### ISSUE TITLE
.iplot() not working with plotly 3.1.1

### ISSUE THREAD

**Issue by mmrahn** [santosjorge/cufflinks#129]
Hi,
I updated plotly from 2.7.0 to 3.1.1. With the following code
```
import pandas as pd
import cufflinks
df = pd.DataFrame(0, columns=[0], index=[0])
df.iplot()
```
I get the following error:
```
C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\_deprecations.py:558: DeprecationWarning:
plotly.graph_objs.YAxis is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.layout.YAxis
  - plotly.graph_objs.layout.scene.YAxis
C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\_deprecations.py:531: DeprecationWarning:
plotly.graph_objs.XAxis is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.layout.XAxis
  - plotly.graph_objs.layout.scene.XAxis
Traceback (most recent call last):
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\IPython\core\interactiveshell.py", line 2963, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-6315a275b9ff>", line 4, in <module>
    df.iplot()
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\cufflinks\plotlytools.py", line 763, in _iplot
    zerolinecolor=zerolinecolor,margin=margin,is3d='3d' in kind,**l_kwargs)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\cufflinks\tools.py", line 201, in getLayout
    layout=go.Layout(theme_data['layout'])
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\_layout.py", line 3737, in __init__
    self.legend = legend if legend is not None else _v
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 3603, in __setattr__
    super(BaseLayoutHierarchyType, self).__setattr__(prop, value)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 2704, in __setattr__
    super(BasePlotlyType, self).__setattr__(prop, value)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\_layout.py", line 1226, in legend
    self['legend'] = val
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 3589, in __setitem__
    super(BaseLayoutHierarchyType, self).__setitem__(prop, value)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 2667, in __setitem__
    self._set_compound_prop(prop, value)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 2965, in _set_compound_prop
    val = validator.validate_coerce(val)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\_plotly_utils\basevalidators.py", line 1792, in validate_coerce
    v = self.data_class(**v)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\layout\_legend.py", line 508, in __init__
    self.font = font if font is not None else _v
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 2704, in __setattr__
    super(BasePlotlyType, self).__setattr__(prop, value)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\layout\_legend.py", line 188, in font
    self['font'] = val
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 2667, in __setitem__
    self._set_compound_prop(prop, value)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 2965, in _set_compound_prop
    val = validator.validate_coerce(val)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\_plotly_utils\basevalidators.py", line 1792, in validate_coerce
    v = self.data_class(**v)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\layout\legend\_font.py", line 209, in __init__
    self.color = color if color is not None else _v
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 2704, in __setattr__
    super(BasePlotlyType, self).__setattr__(prop, value)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\layout\legend\_font.py", line 62, in color
    self['color'] = val
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 2676, in __setitem__
    self._set_prop(prop, value)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 2906, in _set_prop
    val = validator.validate_coerce(val)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\_plotly_utils\basevalidators.py", line 1070, in validate_coerce
    self.raise_invalid_val(v)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\_plotly_utils\basevalidators.py", line 216, in raise_invalid_val
    valid_clr_desc=self.description()))
ValueError: 
    Invalid value of type 'builtins.str' received for the 'color' property of layout.legend.font
        Received value: 'grey10'
    The 'color' property is a color and may be specified as:
      - A hex string (e.g. '#ff0000')
      - An rgb/rgba string (e.g. 'rgb(255,0,0)')
      - An hsl/hsla string (e.g. 'hsl(0,100%,50%)')
      - An hsv/hsva string (e.g. 'hsv(0,100%,100%)')
      - A named CSS color:
            aliceblue, antiquewhite, aqua, aquamarine, azure,
            beige, bisque, black, blanchedalmond, blue,
            blueviolet, brown, burlywood, cadetblue,
            chartreuse, chocolate, coral, cornflowerblue,
            cornsilk, crimson, cyan, darkblue, darkcyan,
            darkgoldenrod, darkgray, darkgrey, darkgreen,
            darkkhaki, darkmagenta, darkolivegreen, darkorange,
            darkorchid, darkred, darksalmon, darkseagreen,
            darkslateblue, darkslategray, darkslategrey,
            darkturquoise, darkviolet, deeppink, deepskyblue,
            dimgray, dimgrey, dodgerblue, firebrick,
            floralwhite, forestgreen, fuchsia, gainsboro,
            ghostwhite, gold, goldenrod, gray, grey, green,
            greenyellow, honeydew, hotpink, indianred, indigo,
            ivory, khaki, lavender, lavenderblush, lawngreen,
            lemonchiffon, lightblue, lightcoral, lightcyan,
            lightgoldenrodyellow, lightgray, lightgrey,
            lightgreen, lightpink, lightsalmon, lightseagreen,
            lightskyblue, lightslategray, lightslategrey,
            lightsteelblue, lightyellow, lime, limegreen,
            linen, magenta, maroon, mediumaquamarine,
            mediumblue, mediumorchid, mediumpurple,
            mediumseagreen, mediumslateblue, mediumspringgreen,
            mediumturquoise, mediumvioletred, midnightblue,
            mintcream, mistyrose, moccasin, navajowhite, navy,
            oldlace, olive, olivedrab, orange, orangered,
            orchid, palegoldenrod, palegreen, paleturquoise,
            palevioletred, papayawhip, peachpuff, peru, pink,
            plum, powderblue, purple, red, rosybrown,
            royalblue, saddlebrown, salmon, sandybrown,
            seagreen, seashell, sienna, silver, skyblue,
            slateblue, slategray, slategrey, snow, springgreen,
            steelblue, tan, teal, thistle, tomato, turquoise,
            violet, wheat, white, whitesmoke, yellow,
            yellowgreen
```
when using cufflinks 0.13.1.

Not sure if this is relevant, but with cufflinks 0.8.2 it does not work either, albeit with a different error:
```
C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\_deprecations.py:558: DeprecationWarning:
plotly.graph_objs.YAxis is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.layout.YAxis
  - plotly.graph_objs.layout.scene.YAxis
C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\_deprecations.py:531: DeprecationWarning:
plotly.graph_objs.XAxis is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.layout.XAxis
  - plotly.graph_objs.layout.scene.XAxis
C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\cufflinks\plotlytools.py:156: FutureWarning:
pandas.tslib is deprecated and will be removed in a future version.
You can access Timestamp as pandas.Timestamp
C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\graph_objs\_deprecations.py:39: DeprecationWarning:
plotly.graph_objs.Data is deprecated.
Please replace it with a list or tuple of instances of the following types
  - plotly.graph_objs.Scatter
  - plotly.graph_objs.Bar
  - plotly.graph_objs.Area
  - plotly.graph_objs.Histogram
  - etc.
Traceback (most recent call last):
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\IPython\core\interactiveshell.py", line 2963, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-6315a275b9ff>", line 4, in <module>
    df.iplot()
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\cufflinks\plotlytools.py", line 901, in _iplot
    figure['data']=data
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 236, in __setitem__
    self.data = value
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 268, in __setattr__
    super(BaseFigure, self).__setattr__(prop, value)
  File "C:\Users\monar\Miniconda3\envs\athion-utils\lib\site-packages\plotly\basedatatypes.py", line 441, in data
    raise ValueError(err_msg)
ValueError: The data property of a figure may only be assigned a list or tuple that contains a permutation of a subset of itself
    Invalid trace(s) with uid(s): {None}
```

I'm using Python 3.6 and Windows 10. It used to work with plotly 2.7.0.

**Comment by timkpaine** [comment]
doesn't work with plotly 3 (look at some of the other issues).

work being done in https://github.com/santosjorge/cufflinks/pull/122

**Comment by santosjorge** [comment]
Fixed in v0.14
